### PR TITLE
REVIEW: NXCM-5181 expose ability to force an update of SystemStatus 

### PR DESCRIPTION
**Pull request by kellyrob99**
https://issues.sonatype.org/browse/NXCM-5181

NXCM-5181 - Can't install new license into expired trial.

Allow for forcing an update of SystemStatus. Needs to be triggered in plugin to solve associated issue.

### CODE REVIEW

**Comment by adreghiciu** [comment]
Started CI build: https://bamboo.zion.sonatype.com/browse/NXO-OSSF48-1

**Comment by adreghiciu** [comment]
+1

**Comment by kellyrob99** [comment]
+1 by CI

**Comment by nabcos** [comment]
Hm, seems like github ate my comment - in short, I would prefer a `void updateSystemStatus()` method instead of `SystemStatus getUpdatedSystemStatus()`, especially because the only caller is not interested in the system status at all.

But that's minor, +0.9 as it is right now ;)

**Comment by kellyrob99** [comment]
@nabcos I can do that, was just trying to preserve the existing semantics for potential future callers, but if we don't have a use case to consume the return value I'm fine with that

**Comment by nabcos** [comment]
IIUC calling update and then getSystemStatus does not have a performance penalty (same status calculation and locking ops) and emphasizes that 'update' is not free (getting license details from PLM means FS roundtrip and decrypting the results)

File: nexus-core/src/main/java/org/sonatype/nexus/AbstractApplicationStatusSource.java [hunk 1]
```diff
@@ -71,19 +71,32 @@ protected ReadWriteLock getLock()
...
     {
-        updateSystemStatusIfNeeded( false );
+        return doGetSystemStatus(false);
+    }
 
-        Lock lock = getLock().readLock();
+    /**
+     * Returns the SystemStatus, forcing an update of possibly cached information, and guaranteeing its consistent state.
+     */
+    public SystemStatus getUpdatedSystemStatus()
```
Comment (nabcos) [inline nexus-core/src/main/java/org/sonatype/nexus/AbstractApplicationStatusSource.java]: should add `@since` on new public API

**Pull request merged**